launch/dynamo-run/src/input/endpoint.rs

```diff
@@ -100,30 +100,28 @@ pub async fn run(
         .await?
         .endpoint(&endpoint_id.name);
 
-    let nats_client = distributed_runtime.nats_client();
-    card.move_to_nats(nats_client.clone()).await?;
-
-    let kvstore: Box<dyn KeyValueStore> =
-        Box::new(NATSStorage::new(nats_client.clone(), endpoint_id));
-    let card_store = Arc::new(KeyValueStoreManager::new(kvstore));
-    card.requires_preprocessing = false;
-    card_store.publish_until_cancelled(
-        cancel_token.clone(),
-        BUCKET_NAME.to_string(),
-        Some(BUCKET_TTL),
-        BUCKET_TTL / 2,
-        card.slug().to_string(),
-        *card.clone(),
-    );
-
     if let Some(etcd_client) = etcd_client {
+        // Store model config files in NATS object store
+        let nats_client = distributed_runtime.nats_client();
+        card.move_to_nats(nats_client.clone()).await?;
+
+        // Publish the Model Deployment Card to etcd
+        let kvstore: Box<dyn KeyValueStore> =
+            Box::new(EtcdStorage::new(etcd_client.clone(), endpoint_id));
+        let card_store = Arc::new(KeyValueStoreManager::new(kvstore));
+        card.requires_preprocessing = false; // TODO this will depend on engine soon
```

Comment:
Is this TODO referring to `EngineConfig::StaticCore` vs `EngineConfig::StaticFull`?

Reply:
Yes. I changed it to make that clearer, which was a necessary next step too.